app/views/users/show.html.erb

```diff
@@ -24,15 +24,103 @@
     }
     <% end %>
   </style>
-  <div class="user-profile-header" style="<%= user_colors_style(@user) %>" itemscope itemtype="http://schema.org/Person" itemprop="mainEntity">
-    <meta itemprop="url" content="https://dev.to/<%= @user.username %>">
+
+<script type="application/ld+json">
+{
+    "@context": "http://schema.org",
+    "@type": "Person",
+    "mainEntityOfPage": {
+      "@type": "WebPage",
+      "@id": "https://dev.to/<%= @user.username %>"
+    },
+    "url": "<%= user_url(@user) %>",
+    "sameAs": [
+      <% if @user.twitter_username? %>
+        "https://twitter.com/<%= @user.twitter_username %>",
+      <% end %>
+      <% if @user.github_username? %>
+        "https://github.com/<%= @user.github_username %>",
+      <% end %>
+      <% if @user.mastodon_url? %>
+        "<%= @user.mastodon_url %>",
+      <% end %>
+      <% if @user.facebook_url? %>
+        "<%= @user.facebook_url %>",
+      <% end %>
+      <% if @user.linkedin_url? %>
+        "<%= @user.linkedin_url %>",
+      <% end %>
+      <% if @user.behance_url? %>
+        "<%= @user.behance_url %>",
+      <% end %>
+      <% if @user.stackoverflow_url? %>
+        "<%= @user.stackoverflow_url %>",
+      <% end %>
+      <% if @user.dribbble_url? %>
+        "<%= @user.dribbble_url %>",
+      <% end %>
+      <% if @user.medium_url? %>
+        "<%= @user.medium_url %>",
+      <% end %>
+      <% if @user.gitlab_url? %>
+        "<%= @user.gitlab_url %>",
+      <% end %>
+      <% if @user.instagram_url? %>
+        "<%= @user.instagram_url %>",
+      <% end %>
+      <% if @user.twitch_username? %>
+        "<%= @user.twitch_url %>",
+      <% end %>
+      <% if @user.website_url? %>
+        "<%= @user.website_url %>"
+      <% end %>
+    ],
+    "image": "<%= ProfileImage.new(@user).get(width: 320) %>",
+    "name": "<%= @user.name %>",
+    <% if @user.email_public == true %>
+      "email": "<%= @user.email %>",
+    <% end %>
+    <% if @user.employment_title.present? %>
+      "jobTitle": "<%= @user.employment_title %>",
+    <% end %>
+    "description": "<%= @user.summary.presence || ["404 bio not found"].sample %>",
+    "disambiguatingDescription": [
+      <% if @user.mostly_work_with.present? %>
+        "<%= @user.mostly_work_with %>",
+      <% end %>
+      <% if @user.currently_hacking_on.present? %>
+        "<%= @user.currently_hacking_on %>",
+      <% end %>
+      <% if @user.currently_learning.present? %>
+        "<%= @user.currently_learning %>"
+      <% end %>
+      ],
+    "worksFor": [
+      {
+        "@type": "Organization",
+        <% if @user.employer_name.present? %>
+          "name": "<%= @user.employer_name %>",
+        <% end %>
+        <% if @user.employer_url.present? %>
+          "sameAs": [ "<%= @user.employer_url %>" ]
+        <% end %>
+      },
+    ],
+    <% if @user.education.present? %>
+      "alumniOf": "<%= @user.education %>"
+    <% end %>
+  }
+</script>
```

Comment:
how did it go with continuing to pull this out into an instance variable and rendering it via the controller? did you run into issues with that?

Reply:
I still ran into a litany of syntax errors, making it not usable for the time being. But, as I mentioned in the comment in the `stories_controller.rb` and in the comment on the PR, I'm still open to trying to get it to work if the consensus is that it is better to extract it into the controller. 😄

Reply:
yeah, i saw the comments, but i was specifically asking because i wasn't sure if you were close to making it work or if it ended up being way more trouble than it was worth. i saw the commented out method but wasn't sure whether you left it in for reference or because you planned to use it later after getting feedback

Reply:
Ah, gotcha! It was definitely a fair amount of trouble, but I left the comment there as both a reference and a shortcut for myself if we decide to go the extraction direction.